### PR TITLE
V3: minor fixes

### PR DESCRIPTION
**Pull request by tuanddd**
## What this PR does
- [x] Modify the Travis CI script so that it will fail on error ([because by default it won't and will try to run all commands](https://docs.travis-ci.com/user/job-lifecycle/#customizing-the-build-phase))
- [x] Fix some filter tests due to the change of `count` and `multi` being true by default
- [x] Fix `sortItems` and `pinItems` tests
- [x] Fix some lint warning, sorry I had my extension disabled without knowing it
- [x] Fix a import error

### CODE REVIEW

**Review by sampotts — APPROVED** [review]

**Review by sampotts — APPROVED** [review]

File: packages/search-ui/src/Filter/utils.ts [hunk 1]
```diff
@@ -8,7 +8,12 @@ export function sortItems(list: Record<string, any>[], prop?: string, asc: boole
   return [...list].sort((a, b) => {
     const l = prop ? a[prop] : a;
     const r = prop ? b[prop] : b;
-    return asc ? l - r : r - l;
+    if (l === r) return 0;
+    if (asc) {
+      if (l > r) return 1;
+      return l > r ? 1 : -1;
```
Comment (sampotts) [inline packages/search-ui/src/Filter/utils.ts:14]: The first line here is redundant, isn't it? It does the same as the first part of the ternary.

File: packages/search-ui/src/Filter/utils.ts [hunk 2]
```diff
@@ -8,7 +8,12 @@ export function sortItems(list: Record<string, any>[], prop?: string, asc: boole
   return [...list].sort((a, b) => {
     const l = prop ? a[prop] : a;
     const r = prop ? b[prop] : b;
-    return asc ? l - r : r - l;
```
Comment (sampotts) [inline packages/search-ui/src/Filter/utils.ts:11]: What was wrong with this logic? AFAIK the sort function just needs to return a negative number, positive number or `0` to determine the sorting.

File: packages/search-ui/src/Filter/utils.ts (hunk 1 above)
Comment (tuanddd) [reply packages/search-ui/src/Filter/utils.ts:14]: Ah yeah my bad.

File: packages/search-ui/src/Filter/utils.ts (hunk 2 above)
Comment (tuanddd) [reply packages/search-ui/src/Filter/utils.ts:11]: Subtracting only works for numbers, if the value is string then you got `NaN` and the array is not sorted.

Comment (sampotts) [reply packages/search-ui/src/Filter/utils.ts:11]: Ah yeah, good point.

Comment (tuanddd) [reply packages/search-ui/src/Filter/utils.ts:11]: I've changed it to be locale dependent using `Intl.Collator`, this is the best option between `localeCompare` and normal comparison ([best speed and has locale support](https://stackoverflow.com/a/52369951))

File: packages/search-ui/src/Filter/utils.ts [hunk 3]
```diff
@@ -4,16 +4,31 @@
...
+  // eslint-disable-next-line no-console
+  console.error(error);
+}
+
 export function sortItems(list: Record<string, any>[], prop?: string, asc: boolean = true) {
   return [...list].sort((a, b) => {
     const l = prop ? a[prop] : a;
     const r = prop ? b[prop] : b;
-    if (l === r) return 0;
+    // Fallback to the normal comparison operator if Collator is not supported;
```
Comment (sampotts) [inline packages/search-ui/src/Filter/utils.ts:19]: Seems a bit complex? Why not something like this:

```javascript
if (typeof l === 'string') {
  const comparison = l.localeCompare(r);
  return asc ? compare : compare * -1; 
}

return asc ? l - r : r - l;
```

Comment (tuanddd) [reply packages/search-ui/src/Filter/utils.ts]: Ah those are not my latest change (forgot to commit them) but anyway let's go with yours :)

**Review by sampotts — APPROVED** [review]

**Pull request merged**